File: GFG/Mathematics/factorial.cpp
#include<iostream>
using namespace std;

int rec_factorial(int n){
    if (n==1){
        return n;
    }
    return n*rec_factorial(n-1);
}

int loop_factorial(int n){
    int fact = 1;
    while (n>0){
        fact=fact*n;
        n--;
    }
    return fact;
}

int main(){
    int x;
    cin>>x;
    cout<<loop_factorial(x);
    return 0;
}

File: GFG/Mathematics/factorial.py
def rec_factorial(n):
    if n==1:
        return n
    return n*rec_factorial(n-1)

def loop_factorial(n):
    fact=1
    while n>=1:
        fact=fact*n
        n-=1
    return fact

print(rec_factorial(25))
print(loop_factorial(20))